src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryTool/DetectBinaries.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.Logging;
+
+namespace BinaryTool;
+public partial class Driver
+{
+    private const string Utf16Marker = "UTF-16";
+
+    // Requires: TargetDirectory exists
+    // Modifies: DetectedBinariesFile
+    // Effects:  Detects binaries in the target directory and writes the results to the DetectedBinariesFile.
+    private void DetectBinaries() => DetectBinariesAsync().GetAwaiter().GetResult();
```

Comment:
Curious, why are we not keeping everything `async`? You can make `Main` async too